src/traces/bar/style_defaults.js

```diff
@@ -23,6 +23,12 @@ module.exports = function handleStyleDefaults(traceIn, traceOut, coerce, default
 
     coerce('marker.line.width');
     coerce('marker.opacity');
+    var pattern = coerce('marker.pattern.shape');
+    if(pattern) {
+        coerce('marker.pattern.bgcolor');
+        coerce('marker.pattern.size');
+        coerce('marker.pattern.solidity');
```

Comment:
We need to add a supplyDefaults test somewhere [here](https://github.com/plotly/plotly.js/blob/ed52205c957630d7a51139235f029a60e728103a/test/jasmine/tests/bar_test.js#L119) for these attributes not be coerced when there is no `pattern.shape`.